### ISSUE TITLE
Damage is quoted in meaningless numbers.

### ISSUE THREAD

**Issue by royfalk** [vegastrike/Assets-Masters#54]
Basic number is VS. This is translated to MJ: x5400

For large weapons, this number is meaningless. Torpedo is 1,080,000,000MJ.
This is 1.08 x 10^12 (unless I miscounted).

According to ~google~ Quora,  1 kT = ~4.18E12 joules.
This comes to 225MT, which is an easier number to understand. It's a really big fusion bomb.

Need to figure out something and modify the assets and scripts accordingly.

**Comment by stephengtuggy** [comment]
That's 1.08 x 10^9, actually.

And yes, it would be very good to use more sensible units of measure across the board. One potential issue, however, is that PWCU and other games might use different units of measure. We might have to make the units of measure part of the data / asset set.

**Comment by BenjamenMeyer** [comment]
> That's 1.08 x 10^9, actually.
> 
> And yes, it would be very good to use more sensible units of measure across the board. One potential issue, however, is that PWCU and other games might use different units of measure. We might have to make the units of measure part of the data / asset set.

We need to define the units of measure in the Engine so the engine can do things accurate. Game Assets (PWCU, VS:UtCS, VSU:BP) should base their units of measure on something the engine publishes.

Perhaps we could setup a conversion utility. One project I had we had a few settings to relate distances and do conversions something like:
- millimeters per sample
- samples per unit

It became easy to convert things to KM, or Miles+Feet or Miles+Chains. We should be able to do something similar here; though at a grander scale; but ultimately it needs to be driven by the engine and configuring the engine.

**Comment by stephengtuggy** [comment]
@BenjamenMeyer Sounds reasonable.

**Comment by royfalk** [comment]
> We need to define the units of measure in the Engine so the engine can do things accurate. Game Assets (PWCU, VS:UtCS, VSU:BP) should base their units of measure on something the engine publishes.

This is out of topic. This is strictly for display of damage in base computer. This code is moving to the assets.
This is also Vegastrike Assets and not engine on purpose. It's game specific.

**Comment by stephengtuggy** [comment]
@royfalk I see. I just created a new ticket to track the other topic separately (the one that @BenjamenMeyer and I were talking about -- defining the units of measure in the Engine).

**Comment by BenjamenMeyer** [comment]
> > We need to define the units of measure in the Engine so the engine can do things accurate. Game Assets (PWCU, VS:UtCS, VSU:BP) should base their units of measure on something the engine publishes.
> 
> This is out of topic. This is strictly for display of damage in base computer. This code is moving to the assets. This is also Vegastrike Assets and not engine on purpose. It's game specific.

Is it? It has to relate to Damage in some form; which the engine manages, not the Assets.

I get it that the display stuff is moving over, as it should; but that also means the Assets and Engine need to be able to talk the same talk with respect to what is getting displayed and what it means.

**Comment by royfalk** [comment]
> Is it? It has to relate to Damage in some form; which the engine manages, not the Assets.

So, we need to separate the underlying numbers from the displayed values.

Ship has 50 points. Missile does 25. These are arbitrary game numbers.

Ship can sustain 6.7 MJ or 1.21 Gwatt are display numbers and can vary between games. It's all funny numbers anyhow.

This is also why it makes less sense to do this for the physics stuff, which are grounded in reality AFAIK.

Right now we use tons, meters, seconds etc. We could convert them to miles but this or maybe decimal time but this would require some careful changes to the engine.

**Comment by BenjamenMeyer** [comment]
@royfalk agreed - we need consistency; and to make things meaningful in terms of the game play from the assets.
